src/cpu/o3/trace/TraceFetch.cc

```
@@ -0,0 +1,1605 @@
+        // Note: pass CPU as parent stats group; keep reader name simple to avoid
+        // duplicated prefixes in stats paths.
+        traceReader = createTraceReader(params.traceFormat, params.traceFile,
+                                        "traceReader",
+                                        params.traceAddrBase, params.traceAddrSize,
+                                        params.traceAddrMapMode, params.traceAddrPageAlign,
+                                        fetch.cpu);
+        if (!traceReader) {
+            fatal("Failed to create trace reader for format: %s\n",
+                  params.traceFormat);
+        }
+    } else {
```

Cppcheck (2.19.0)

[error] Reference to temporary returned.

(returnTempReference)